### PR TITLE
second exercise

### PR DESCRIPTION
**Pull request by itayl13**

### CODE REVIEW

File: ex2/ex_2.cs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+     */
+    class Answers
+    {
+        // 1
+        public enum AccountType { Checking, Deposit }
+
+        // 2
+        public struct BankAccount : IComparable<BankAccount>
+        { 
+            public string owner; public int moneyInAccount; public AccountType accType;
```
Comment (ofirbarak) [inline ex2/ex_2.cs]: Separate each one to a new line.  
Note for the access modifiers and naming conventions (public should be Pascal case)  https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties

File: ex2/ex_2.cs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                    moneyInAccount = acc.moneyInAccount + money,
+                    accType = acc.accType
+                };
+                return newAcc;
+            }
+
+        }
+
+        // Helping function to print BankAccount elements, a common action.
+        static void PrintAccount(BankAccount acc)
```
Comment (ofirbarak) [inline ex2/ex_2.cs]: Instead override the `ToString` method.

File: ex2/ex_2.cs [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+     *    order.
+     * 6. Write a method called AddMoney which receives a BankAccount argument by reference, as well as an integer 
+     *    representing an amount of money, that will add to the BankAccount amount of money this integer amount.
+     * 7. Add to BankAccount the operator + so we could use it to directly update the amount of money in the account.
+     * 8. Write a method called IsPalindrome which receives an integer and checks whether it is a palindrome.
+     */
+    class Answers
+    {
+        // 1
+        public enum AccountType { Checking, Deposit }
```
Comment (ofirbarak) [inline ex2/ex_2.cs]: Should it be public?

File: ex2/ex_2.cs [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        {
+            Console.WriteLine("Owner: {0}\nMoney In Account: {1}\nAccount Type: {2}",
+                acc.owner, acc.moneyInAccount, acc.accType);
+        }
+
+        // 4
+        static int SumOfBalances(params BankAccount[] accounts)
+        {
+            int sum = 0;
+            foreach (BankAccount acc in accounts)
```
Comment (ofirbarak) [inline ex2/ex_2.cs]: what happens when `accounts == null`

Comment (ofirbarak) [reply ex2/ex_2.cs]: Don't short on names - `acc` should be `account` or something similar. Think on someone new that reads this code in the first time of his life, he should not guess what you meant.

File: ex2/ex_2.cs [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+            {
+                sum += acc.moneyInAccount;
+            }
+            return sum;
+        }
+
+        // 5
+        static void GetRichestAccount(List<BankAccount> accounts)
```
Comment (ofirbarak) [inline ex2/ex_2.cs]: It called `GetRichestAccount` but returns nothing

File: ex2/ex_2.cs [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+        {
+            string strnum = num.ToString();
+            for (int i=0; i < (strnum.Length + 1) / 2; i++)
+            {
+                if (!(strnum[i] == strnum[strnum.Length - i - 1])) return false;
+            }
+            return true;
+        }
+
+        static void Main()
```
Comment (ofirbarak) [inline ex2/ex_2.cs:91]: `Main` should be in another class, usually `Program`.

File: ex2/ex_2.cs [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+            }
+            return sum;
+        }
+
+        // 5
+        static void GetRichestAccount(List<BankAccount> accounts)
+        {
+            BankAccount richest = accounts.Max();
+            Console.WriteLine("The richest account:");
+            PrintAccount(richest);
```
Comment (ofirbarak) [inline ex2/ex_2.cs]: Every function does one thing. Print with `ToString` method always. And do it in the `Main`, what if tomorrow you'll decide printing is a weirdo thing.

File: ex2/ex_2.cs [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+        }
+        static void GetPoorestAccount(List<BankAccount> accounts)
```
Comment (ofirbarak) [inline ex2/ex_2.cs]: Same as above

File: ex2/ex_2.cs [hunk 9]
```diff
@@ -19,68 +19,63 @@ namespace Ex2
...
         // 1
-        public enum AccountType { Checking, Deposit }
+        enum AccountType { Checking, Deposit }
 
         // 2
-        public struct BankAccount : IComparable<BankAccount>
+        struct BankAccount : IComparable<BankAccount>
         { 
-            public string owner; public int moneyInAccount; public AccountType accType; 
+            internal string owner;
```
Comment (ofirbarak) [inline ex2/ex_2.cs:30]: Why `internal` and not `public`? Do you know the difference?

File: ex2/ex_2.cs [hunk 10]
```diff
@@ -97,56 +92,63 @@ static void Main()
...
             // Test 6
-            AddMoney(ref account, 10);
-            PrintAccount(account);
+            AddMoney(ref account0, 10);
+            Console.WriteLine(account0.ToString());
             // Test 7
-            account += 10;
-            PrintAccount(account);
+            account0 += 10;
+            Console.WriteLine(account0.ToString());
```
Comment (ofirbarak) [inline ex2/ex_2.cs]: You do not have to explicitly call `ToString` here, the function `Console.WriteLine` does in implicitly. Read the guide of casting https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/casting-and-type-conversions

File: ex2/ex_2.cs (hunk 9 above)
Comment (itayl13) [reply ex2/ex_2.cs:30]: I know the difference. The question of accessibility to these fields is not relevant to the questions in the exercise, both access modifiers will give the same results to the relevant questions. 
To solve the questions, the lowest level of accessibility necessary is `internal`, but you probably see the role of `BankAccount` struct different than how I do. If you think these should be `public`, I will change them.

**Pull request merged**